client-runtime/src/main/java/com/microsoft/rest/v3/http/rest/SimpleRestResponse.java

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.microsoft.rest.v3.http.rest;
+
+import com.microsoft.rest.v3.http.HttpHeaders;
+import com.microsoft.rest.v3.http.HttpRequest;
+
+import java.util.Map;
+
+/**
+ * REST response with a strongly-typed content specified.
+ *
+ * @param <T> The deserialized type of the response content.
+ */
+public class SimpleRestResponse<T> implements RestResponse<T> {
```

Comment:
Was initially thinking why we need `SimpleRestResponse` class type given we support returning `RestResponse` interface type. Thinking more on this, it make sense to have `SimpleRestResponse`, e.g. SDKs returning their own response type with no custom/deserialized headers can extend from this type instead of RestResponseBase. This way those response types don't endup with  Void custom headers.

Reply:
Yes, I've always been on the fence about this, but I think it makes sense still.